lib/Service/RoomService.php

```diff
@@ -993,6 +1003,154 @@ public function setLastActivity(Room $room, \DateTime $now): void {
+					$changed[] = ARoomModifiedEvent::PROPERTY_DESCRIPTION;
+				}
+			} catch (\LengthException $e) {
+				$this->logger->error('A \LengthException occurred while trying to sync description of ' . $local->getId() . ' to ' . $host['description'], ['exception' => $e]);
+			}
+		}
+		if (isset($host['callRecording']) && $host['callRecording'] !== $local->getCallRecording()) {
+			try {
+				$this->setCallRecording($local, $host['callRecording']);
+				$changed[] = ARoomModifiedEvent::PROPERTY_CALL_RECORDING;
+			} catch (\InvalidArgumentException $e) {
+				$this->logger->error('An error (' . $e->getMessage() . ') occurred while trying to sync callRecording of ' . $local->getId() . ' to ' . $host['callRecording'], ['exception' => $e]);
+			}
+		}
+		if (isset($host['defaultPermissions']) && $host['defaultPermissions'] !== $local->getDefaultPermissions()) {
+			$success = $this->setPermissions($local, 'default', Attendee::PERMISSIONS_MODIFY_SET, $host['defaultPermissions'], false);
+			if (!$success) {
+				$this->logger->error('An error occurred while trying to sync defaultPermissions of ' . $local->getId() . ' to ' . $host['defaultPermissions']);
+			} else {
+				$changed[] = ARoomModifiedEvent::PROPERTY_DEFAULT_PERMISSIONS;
+			}
+		}
+		if (isset($host['avatarVersion']) && $host['avatarVersion'] !== $local->getAvatar()) {
+			$hostAvatar = $host['avatarVersion'];
+			if ($hostAvatar) {
+				// Add a fake suffix as we explode by the dot in the AvatarService, but the version doesn't have one.
+				$hostAvatar .= '.fed';
+			}
+			$success = $this->setAvatar($local, $hostAvatar);
+			if (!$success) {
+				$this->logger->error('An error occurred while trying to sync avatarVersion of ' . $local->getId() . ' to ' . $host['avatarVersion']);
+			} else {
+				$changed[] = ARoomModifiedEvent::PROPERTY_AVATAR;
+			}
+		}
+		if (isset($host['lastActivity']) && $host['lastActivity'] !== 0 && $host['lastActivity'] !== ((int) $local->getLastActivity()?->getTimestamp())) {
+			$lastActivity = $this->timeFactory->getDateTime('@' . $host['lastActivity']);
+			$this->setLastActivity($local, $lastActivity);
+			$changed[] = ARoomSyncedEvent::PROPERTY_LAST_ACTIVITY;
+		}
+		if (isset($host['lobbyState'], $host['lobbyTimer']) && ($host['lobbyState'] !== $local->getLobbyState(false) || $host['lobbyTimer'] !== ((int) $local->getLobbyTimer(false)?->getTimestamp()))) {
+			$hostTimer = $host['lobbyTimer'] === 0 ? null : $this->timeFactory->getDateTime('@' . $host['lobbyTimer']);
+			$success = $this->setLobby($local, $host['lobbyState'], $hostTimer);
+			if (!$success) {
+				$this->logger->error('An error occurred while trying to sync lobby of ' . $local->getId() . ' to ' . $host['lobbyState'] . ' with timer to ' . $host['lobbyTimer']);
+			} else {
+				$changed[] = ARoomModifiedEvent::PROPERTY_LOBBY;
+			}
+		}
+		if (isset($host['callStartTime'], $host['callFlag'])) {
+			if ($host['callStartTime'] === 0 || $host['callFlag'] === Participant::FLAG_DISCONNECTED) {
```

Comment:
The `callFlag` and `callStartTime` are always set when joining the conversation even if there was no call before, as the current value is not taken into account, only if the values are provided in the data, which always are.

But this may be a bit tricky to properly update, it seems that when joining again `$local->getCallFlag` is always disconnected :thinking:

Reply:
I think my brain had too much boolean logic... Fixed now